### PR TITLE
[Facility Locator] Fixes for pagination issues

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR fixes the pagination of the results list on the Facility Locator, which currently hits a runtime error. The total-results count was also wrong.

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17247

### Error
image

## Testing done
Local testing pointed at the Dev API

### Page 1 (Cincinnati)
image

### Page 2 (Cincinnati)
image

## Acceptance criteria
- [ ] Pagination works

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/actions/index.js [hunk 1]
```diff
@@ -227,6 +227,7 @@ export const genBBoxFromAddress = (query) => {
               },
               bounds: minBounds,
               zoomLevel: res.features[0].id.split('.')[0] === 'region' ? 7 : 9,
+              currentPage: 1,
```
Comment (ncksllvn) [inline src/applications/facility-locator/actions/index.js]: The idea here is that if the locations list needs to be updated based on a map-change (zoom in or lat/long), then the current page should revert to 1.

File: src/applications/facility-locator/components/ResultsList.jsx [hunk 2]
```diff
@@ -152,6 +152,7 @@ function mapStateToProps(state) {
   const facilityTypeName = facilityTypes[facilityType];
 
   return {
+    currentQuery: state.searchQuery,
```
Comment (ncksllvn) [inline src/applications/facility-locator/components/ResultsList.jsx]: This was previously removed, but its removal was causing a runtime error so I'm adding it back.

File: src/applications/facility-locator/components/ResultsList.jsx [hunk 3]
```diff
@@ -99,7 +99,7 @@ class ResultsList extends Component {
         {/* eslint-disable jsx-a11y/no-noninteractive-tabindex */}
         <p className="search-result-title" ref={this.searchResultTitle}>
           {/* eslint-enable jsx-a11y/no-noninteractive-tabindex */}
-          {`${results.length} results for ${facilityTypeName} near `}
+          {`${totalEntries} results for ${facilityTypeName} near `}
```
Comment (ncksllvn) [inline src/applications/facility-locator/components/ResultsList.jsx]: `totalEntries` is the complete count of search results, whereas `results.length` is the count of the current page. `totalEntries` makes more sense.

**Review by jbalboni — APPROVED** [review]

**Review by bshyong — APPROVED** [review]
LGTM

File: src/applications/facility-locator/components/ResultsList.jsx (hunk 2 above)
Comment (bshyong) [reply src/applications/facility-locator/components/ResultsList.jsx]: this might be used for keeping track of query state and pushing it to the URL

**Pull request merged**